src/metabase/api/collection.clj

```diff
@@ -349,26 +349,53 @@
 (defn- collection-children*
   [collection models {:keys [sort-info] :as options}]
   (let [sql-order   (case sort-info
-                      nil                  [[:%lower.name :asc]]
-                      [:name :asc]         [[:%lower.name :asc]]
-                      [:name :desc]        [[:%lower.name :desc]]
-                      [:last-edited :asc]  [(if (= @mdb.env/db-type :mysql)
-                                              [(hsql/call :ISNULL :last_edit_timestamp)]
-                                              [:last_edit_timestamp :nulls-last])
-                                            [:last_edit_timestamp :asc]
-                                            [:%lower.name :asc]]
-                      [:last-edited :desc] (into
-                                            (case @mdb.env/db-type
-                                              :mysql
-                                              [[(hsql/call :ISNULL :last_edit_timestamp)]]
-                                              :postgres
-                                              [[(hsql/raw "last_edit_timestamp DESC NULLS LAST")]]
-                                              :h2
-                                              [])
-                                            [[:last_edit_timestamp :desc]
-                                             [:%lower.name :asc]])
-                      [:model :asc]        [[:model_ranking :asc]  [:%lower.name :asc]]
-                      [:model :desc]       [[:model_ranking :desc] [:%lower.name :asc]])
+                      nil                     [[:%lower.name :asc]]
+                      [:name :asc]            [[:%lower.name :asc]]
+                      [:name :desc]           [[:%lower.name :desc]]
+                      [:last-edited-at :asc]  [(if (= @mdb.env/db-type :mysql)
+                                                 [(hsql/call :ISNULL :last_edit_timestamp)]
+                                                 [:last_edit_timestamp :nulls-last])
+                                               [:last_edit_timestamp :asc]
+                                               [:%lower.name :asc]]
```

Comment:
a lot of work to ensure that nulls always go last. Depends on the db, and we don't have the benefit from honeysql 2 which extends the order-by grammar allowing for `[:name :desc :nulls :last]` or whatever the syntax is.

Reply:
is there any possibility to make this more explicit by making a function do this legerdemain with the IS_NULLs instead of having a buncha ifs inside of a big ol' case?

Reply:
(this would also get rid of the other thing I find annoying about this, which is mere size: what an absolute unit of a case)